addons/account/models/account_move.py

```diff
@@ -3899,7 +3899,7 @@ def _compute_amount_residual(self):
                 line.reconciled = (
                     line.company_currency_id.is_zero(line.amount_residual)
                     and (not line.currency_id or line.currency_id.is_zero(line.amount_residual_currency))
-                    and line.move_id.state not in ('draft', 'cancel')
+                    and (line.matched_debit_ids or line.matched_credit_ids)
```

Comment:
@william-andre this solves your issue.
ATM, 0 balance line are always considered reconciled and they never have `line.matched_debit_ids or line.matched_credit_ids`
```suggestion
                    and (line.matched_debit_ids or line.matched_credit_ids or line.company_currency_id.is_zero(line.balance))
```

Reply:
At first glance there should be one (and only one) partial, even for an amount of 0

Reply:
Here https://github.com/odoo/odoo/pull/97037 I made the complete fix. Fell free to cheery-pick and customize it :)